src/frequenz/client/dispatch/test/_service.py

```diff
@@ -50,12 +51,36 @@
 class FakeService:
     """Dispatch mock service for testing."""
 
+    @dataclass(frozen=True)
+    class StreamEvent:
+        """Event for the stream."""
+
+        microgrid_id: int
+        """The microgrid id."""
+
+        event: DispatchEvent
+        """The event."""
+
+    _stream_channel: Broadcast[StreamEvent]
+    """Channel for dispatch events."""
+
+    _stream_sender: Sender[StreamEvent]
+    """Sender for dispatch events."""
+
     dispatches: dict[int, list[Dispatch]] = dataclasses.field(default_factory=dict)
     """List of dispatches per microgrid."""
 
     _last_id: int = 0
     """Last used dispatch id."""
 
+    def __init__(self) -> None:
+        """Initialize the service."""
+        super().__init__()
+        self._last_id = 0
+        self._stream_channel = Broadcast(name="dispatch-stream")
+        self._stream_sender = self._stream_channel.new_sender()
+        self.dispatches = {}
```

Comment:
If you have this, it makes no sense to have the `default_factory` in `dispatches: dict[int, list[Dispatch]] = dataclasses.field(default_factory=dict)`.

I would add a `default_factory` for the channel too instead of implementing `__init__()` though.